core/benches/receive_and_buffer.rs

```diff
@@ -0,0 +1,278 @@
+                        is_writable: true,
+                    }],
+                ));
+            }
+            VersionedTransaction::try_new(
+                VersionedMessage::Legacy(Message::new_with_blockhash(
+                    &instructions,
+                    Some(&fee_payer.pubkey()),
+                    &blockhash.get(&mut rng),
+                )),
+                &[&fee_payer],
+            )
+            .unwrap()
+        })
+        .collect();
+
+    let packets_batches = BankingPacketBatch::new(to_packet_batches(&txs, NUM_PACKETS));
+    return packets_batches;
+}
+
+// TODO(klykov): maybe not needed anylonger if add new to the TransactionViewReceiveAndBuffer?
+trait ReceiveAndBufferCreator {
+    fn create(
+        receiver: Receiver<Arc<Vec<PacketBatch>>>,
+        bank_forks: Arc<RwLock<BankForks>>,
+    ) -> Self;
+}
+
+impl ReceiveAndBufferCreator for TransactionViewReceiveAndBuffer {
+    fn create(
+        receiver: Receiver<Arc<Vec<PacketBatch>>>,
+        bank_forks: Arc<RwLock<BankForks>>,
+    ) -> Self {
+        TransactionViewReceiveAndBuffer {
+            receiver,
+            bank_forks,
+        }
+    }
+}
+
+impl ReceiveAndBufferCreator for SanitizedTransactionReceiveAndBuffer {
+    fn create(
+        receiver: Receiver<Arc<Vec<PacketBatch>>>,
+        bank_forks: Arc<RwLock<BankForks>>,
+    ) -> Self {
+        SanitizedTransactionReceiveAndBuffer::new(PacketDeserializer::new(receiver), bank_forks)
+    }
+}
+
+fn bench_receive_and_buffer<T: ReceiveAndBuffer + ReceiveAndBufferCreator>(
+    c: &mut Criterion,
+    bench_name: &str,
+    num_instructions_per_tx: usize,
+    probability_invalid_blockhash: f64,
+) {
+    let GenesisConfigInfo {
+        mut genesis_config, ..
+    } = create_genesis_config(100_000);
+    let num_txs = 16 * 1024;
+    // This doesn't change the time to execute
+    let num_fee_payers = 1;
+    // fee payers will be verified, so have to create them properly
+    let fee_payers = create_accounts(num_fee_payers, &mut genesis_config);
+
+    let (bank, bank_forks) =
+        Bank::new_for_benches(&genesis_config).wrap_with_bank_forks_for_tests();
+    let bank_start = BankStart {
+        working_bank: bank.clone(),
+        bank_creation_time: Arc::new(Instant::now()),
+    };
+
+    let (sender, receiver) = unbounded();
+
+    let mut rb = T::create(receiver, bank_forks);
+
+    const TOTAL_BUFFERED_PACKETS: usize = 100_000;
+    let mut count_metrics = SchedulerCountMetrics::default();
+    let mut timing_metrics = SchedulerTimingMetrics::default();
+    let decision = BufferedPacketsDecision::Consume(bank_start);
+
+    let txs = generate_transactions(
+        num_txs,
+        bank.clone(),
+        &fee_payers,
+        num_instructions_per_tx,
+        probability_invalid_blockhash,
+    );
+    let mut container = <T as ReceiveAndBuffer>::Container::with_capacity(TOTAL_BUFFERED_PACKETS);
+    c.bench_function(bench_name, |bencher| {
```

Comment:
Can we add throughput measurements in terms of transactions? It gives us a nice number to report in terms of TPS capacity.

See: https://bheisler.github.io/criterion.rs/book/user_guide/advanced_configuration.html#throughput-measurements